Pyinstaller broken for macOS

MacOS is disabled in the GH action 'release' due to the following error:
```
2023-11-14T19:49:11.4080320Z Processing /Users/runner/work/minegauler/minegauler
2023-11-14T19:49:11.4141160Z   Installing build dependencies: started
2023-11-14T19:49:14.0134520Z   Installing build dependencies: finished with status 'done'
2023-11-14T19:49:14.0152830Z   Getting requirements to build wheel: started
2023-11-14T19:49:14.2998150Z   Getting requirements to build wheel: finished with status 'done'
2023-11-14T19:49:14.3035740Z   Preparing metadata (pyproject.toml): started
2023-11-14T19:49:14.5752270Z   Preparing metadata (pyproject.toml): finished with status 'done'
2023-11-14T19:49:14.7565880Z Collecting pyinstaller
2023-11-14T19:49:14.7570050Z   Obtaining dependency information for pyinstaller from https://files.pythonhosted.org/packages/6b/3d/ac556c1208c701c4f6ccde77db0b2eeb84bf052d63260c2305fb5ace707c/pyinstaller-6.2.0-py3-none-macosx_10_13_universal2.whl.metadata
2023-11-14T19:49:14.8044190Z   Downloading pyinstaller-6.2.0-py3-none-macosx_10_13_universal2.whl.metadata (8.3 kB)
2023-11-14T19:49:14.8290680Z Requirement already satisfied: setuptools>=42.0.0 in /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages (from pyinstaller) (58.1.0)
2023-11-14T19:49:14.8631020Z Collecting altgraph (from pyinstaller)
2023-11-14T19:49:14.8636230Z   Obtaining dependency information for altgraph from https://files.pythonhosted.org/packages/4d/3f/3bc3f1d83f6e4a7fcb834d3720544ca597590425be5ba9db032b2bf322a2/altgraph-0.17.4-py2.py3-none-any.whl.metadata
2023-11-14T19:49:14.8791670Z   Downloading altgraph-0.17.4-py2.py3-none-any.whl.metadata (7.3 kB)
2023-11-14T19:49:14.9437730Z Collecting pyinstaller-hooks-contrib>=2021.4 (from pyinstaller)
2023-11-14T19:49:14.9443380Z   Obtaining dependency information for pyinstaller-hooks-contrib>=2021.4 from https://files.pythonhosted.org/packages/2b/5f/30224db46281199f405b9c3b929f43a7d17591180b52f13c75c625dfc386/pyinstaller_hooks_contrib-2023.10-py2.py3-none-any.whl.metadata
2023-11-14T19:49:14.9603340Z   Downloading pyinstaller_hooks_contrib-2023.10-py2.py3-none-any.whl.metadata (15 kB)
2023-11-14T19:49:15.0210480Z Collecting packaging>=22.0 (from pyinstaller)
2023-11-14T19:49:15.0258540Z   Obtaining dependency information for packaging>=22.0 from https://files.pythonhosted.org/packages/ec/1a/610693ac4ee14fcdf2d9bf3c493370e4f2ef7ae2e19217d7a237ff42367d/packaging-23.2-py3-none-any.whl.metadata
2023-11-14T19:49:15.0367570Z   Downloading packaging-23.2-py3-none-any.whl.metadata (3.2 kB)
2023-11-14T19:49:15.1533170Z Collecting importlib-metadata>=4.6 (from pyinstaller)
2023-11-14T19:49:15.1536920Z   Obtaining dependency information for importlib-metadata>=4.6 from https://files.pythonhosted.org/packages/cc/37/db7ba97e676af155f5fcb1a35466f446eadc9104e25b83366e8088c9c926/importlib_metadata-6.8.0-py3-none-any.whl.metadata
2023-11-14T19:49:15.1683010Z   Downloading importlib_metadata-6.8.0-py3-none-any.whl.metadata (5.1 kB)
2023-11-14T19:49:15.2134950Z Collecting macholib>=1.8 (from pyinstaller)
2023-11-14T19:49:15.2140090Z   Obtaining dependency information for macholib>=1.8 from https://files.pythonhosted.org/packages/d1/5d/c059c180c84f7962db0aeae7c3b9303ed1d73d76f2bfbc32bc231c8be314/macholib-1.16.3-py2.py3-none-any.whl.metadata
2023-11-14T19:49:15.2307960Z   Downloading macholib-1.16.3-py2.py3-none-any.whl.metadata (11 kB)
2023-11-14T19:49:15.2893650Z Collecting attrs (from minegauler==4.2.0b0)
2023-11-14T19:49:15.3043270Z   Downloading attrs-23.1.0-py3-none-any.whl (61 kB)
2023-11-14T19:49:15.3199380Z      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.2/61.2 kB 4.2 MB/s eta 0:00:00
2023-11-14T19:49:15.3925620Z Collecting PyQt5 (from minegauler==4.2.0b0)
2023-11-14T19:49:15.3930860Z   Obtaining dependency information for PyQt5 from https://files.pythonhosted.org/packages/f2/8a/aa7699436ed898a1cc34724bbb672a879ea51b8067f0ab71843d52f6fcd0/PyQt5-5.15.10-cp37-abi3-macosx_10_13_x86_64.whl.metadata
2023-11-14T19:49:15.4094170Z   Downloading PyQt5-5.15.10-cp37-abi3-macosx_10_13_x86_64.whl.metadata (2.1 kB)
2023-11-14T19:49:15.4923980Z Collecting requests (from minegauler==4.2.0b0)
2023-11-14T19:49:15.4927890Z   Obtaining dependency information for requests from https://files.pythonhosted.org/packages/70/8e/0e2d847013cb52cd35b38c009bb167a1a26b2ce6cd6965bf26b47bc0bf44/requests-2.31.0-py3-none-any.whl.metadata
2023-11-14T19:49:15.5068300Z   Downloading requests-2.31.0-py3-none-any.whl.metadata (4.6 kB)
2023-11-14T19:49:15.5560980Z Collecting typing-extensions (from minegauler==4.2.0b0)
2023-11-14T19:49:15.5563870Z   Obtaining dependency information for typing-extensions from https://files.pythonhosted.org/packages/24/21/7d397a4b7934ff4028987914ac1044d3b7d52712f30e2ac7a2ae5bc86dd0/typing_extensions-4.8.0-py3-none-any.whl.metadata
2023-11-14T19:49:15.5709480Z   Downloading typing_extensions-4.8.0-py3-none-any.whl.metadata (3.0 kB)
2023-11-14T19:49:15.6422900Z Collecting zig-minesolver<0.2,>=0.1.1 (from minegauler==4.2.0b0)
2023-11-14T19:49:15.6614440Z   Downloading zig_minesolver-0.1.1-py3-none-macosx_10_9_x86_64.whl (174 kB)
2023-11-14T19:49:15.6756860Z      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 174.8/174.8 kB 14.9 MB/s eta 0:00:00
2023-11-14T19:49:15.7804670Z Collecting pytz (from minegauler==4.2.0b0)
2023-11-14T19:49:15.7807930Z   Obtaining dependency information for pytz from https://files.pythonhosted.org/packages/32/4d/aaf7eff5deb402fd9a24a1449a8119f00d74ae9c2efa79f8ef9994261fc2/pytz-2023.3.post1-py2.py3-none-any.whl.metadata
2023-11-14T19:49:15.7953360Z   Downloading pytz-2023.3.post1-py2.py3-none-any.whl.metadata (22 kB)
2023-11-14T19:49:15.8397890Z Collecting requests-toolbelt (from minegauler==4.2.0b0)
2023-11-14T19:49:15.8542240Z   Downloading requests_toolbelt-1.0.0-py2.py3-none-any.whl (54 kB)
2023-11-14T19:49:15.8624870Z      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.5/54.5 kB 8.4 MB/s eta 0:00:00
2023-11-14T19:49:15.8987620Z Collecting tabulate (from minegauler==4.2.0b0)
2023-11-14T19:49:15.9130570Z   Downloading tabulate-0.9.0-py3-none-any.whl (35 kB)
2023-11-14T19:49:15.9976260Z Collecting zipp>=0.5 (from importlib-metadata>=4.6->pyinstaller)
2023-11-14T19:49:15.9978800Z   Obtaining dependency information for zipp>=0.5 from https://files.pythonhosted.org/packages/d9/66/48866fc6b158c81cc2bfecc04c480f105c6040e8b077bc54c634b4a67926/zipp-3.17.0-py3-none-any.whl.metadata
2023-11-14T19:49:16.0121900Z   Downloading zipp-3.17.0-py3-none-any.whl.metadata (3.7 kB)
2023-11-14T19:49:16.2205840Z Collecting PyQt5-sip<13,>=12.13 (from PyQt5->minegauler==4.2.0b0)
2023-11-14T19:49:16.2209450Z   Obtaining dependency information for PyQt5-sip<13,>=12.13 from https://files.pythonhosted.org/packages/cf/00/442b0fd2ae8d88eece790856b71b35f2a0cf467e63d7adfc4512cd50ef44/PyQt5_sip-12.13.0-cp39-cp39-macosx_10_9_universal2.whl.metadata
2023-11-14T19:49:16.2375340Z   Downloading PyQt5_sip-12.13.0-cp39-cp39-macosx_10_9_universal2.whl.metadata (504 bytes)
2023-11-14T19:49:16.2680010Z Collecting PyQt5-Qt5>=5.15.2 (from PyQt5->minegauler==4.2.0b0)
2023-11-14T19:49:16.2683610Z   Obtaining dependency information for PyQt5-Qt5>=5.15.2 from https://files.pythonhosted.org/packages/66/f5/0a684b2b33b1f1dceaf894d4e8a105f523bb08003f81c89a91e509c5f854/PyQt5_Qt5-5.15.11-py3-none-macosx_10_13_x86_64.whl.metadata
2023-11-14T19:49:16.2830130Z   Downloading PyQt5_Qt5-5.15.11-py3-none-macosx_10_13_x86_64.whl.metadata (536 bytes)
2023-11-14T19:49:16.5649290Z Collecting charset-normalizer<4,>=2 (from requests->minegauler==4.2.0b0)
2023-11-14T19:49:16.5871450Z   Obtaining dependency information for charset-normalizer<4,>=2 from https://files.pythonhosted.org/packages/66/fe/c7d3da40a66a6bf2920cce0f436fa1f62ee28aaf92f412f0bf3b84c8ad6c/charset_normalizer-3.3.2-cp39-cp39-macosx_10_9_x86_64.whl.metadata
2023-11-14T19:49:16.5886120Z   Downloading charset_normalizer-3.3.2-cp39-cp39-macosx_10_9_x86_64.whl.metadata (33 kB)
2023-11-14T19:49:16.6221740Z Collecting idna<4,>=2.5 (from requests->minegauler==4.2.0b0)
2023-11-14T19:49:16.6364130Z   Downloading idna-3.4-py3-none-any.whl (61 kB)
2023-11-14T19:49:16.6452890Z      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.5/61.5 kB 8.2 MB/s eta 0:00:00
2023-11-14T19:49:16.7328100Z Collecting urllib3<3,>=1.21.1 (from requests->minegauler==4.2.0b0)
2023-11-14T19:49:16.7331250Z   Obtaining dependency information for urllib3<3,>=1.21.1 from https://files.pythonhosted.org/packages/96/94/c31f58c7a7f470d5665935262ebd7455c7e4c7782eb525658d3dbf4b9403/urllib3-2.1.0-py3-none-any.whl.metadata
2023-11-14T19:49:16.7481890Z   Downloading urllib3-2.1.0-py3-none-any.whl.metadata (6.4 kB)
2023-11-14T19:49:16.7997200Z Collecting certifi>=2017.4.17 (from requests->minegauler==4.2.0b0)
2023-11-14T19:49:16.8000090Z   Obtaining dependency information for certifi>=2017.4.17 from https://files.pythonhosted.org/packages/4c/dd/2234eab22353ffc7d94e8d13177aaa050113286e93e7b40eae01fbf7c3d9/certifi-2023.7.22-py3-none-any.whl.metadata
2023-11-14T19:49:16.8047350Z   Using cached certifi-2023.7.22-py3-none-any.whl.metadata (2.2 kB)
2023-11-14T19:49:16.9141190Z Downloading pyinstaller-6.2.0-py3-none-macosx_10_13_universal2.whl (939 kB)
2023-11-14T19:49:16.9489500Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 940.0/940.0 kB 29.9 MB/s eta 0:00:00
2023-11-14T19:49:16.9633010Z Downloading importlib_metadata-6.8.0-py3-none-any.whl (22 kB)
2023-11-14T19:49:16.9866810Z Downloading macholib-1.16.3-py2.py3-none-any.whl (38 kB)
2023-11-14T19:49:17.0064280Z Downloading altgraph-0.17.4-py2.py3-none-any.whl (21 kB)
2023-11-14T19:49:17.0360730Z Downloading packaging-23.2-py3-none-any.whl (53 kB)
2023-11-14T19:49:17.0447980Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.0/53.0 kB 7.1 MB/s eta 0:00:00
2023-11-14T19:49:17.0637440Z Downloading pyinstaller_hooks_contrib-2023.10-py2.py3-none-any.whl (288 kB)
2023-11-14T19:49:17.0774060Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 288.4/288.4 kB 30.1 MB/s eta 0:00:00
2023-11-14T19:49:17.1142370Z Downloading PyQt5-5.15.10-cp37-abi3-macosx_10_13_x86_64.whl (7.0 MB)
2023-11-14T19:49:17.3353310Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.0/7.0 MB 32.0 MB/s eta 0:00:00
2023-11-14T19:49:17.3541130Z Downloading pytz-2023.3.post1-py2.py3-none-any.whl (502 kB)
2023-11-14T19:49:17.3858040Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 502.5/502.5 kB 18.2 MB/s eta 0:00:00
2023-11-14T19:49:17.4007650Z Downloading requests-2.31.0-py3-none-any.whl (62 kB)
2023-11-14T19:49:17.4122760Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.6/62.6 kB 6.2 MB/s eta 0:00:00
2023-11-14T19:49:17.4265100Z Downloading typing_extensions-4.8.0-py3-none-any.whl (31 kB)
2023-11-14T19:49:17.4380450Z Using cached certifi-2023.7.22-py3-none-any.whl (158 kB)
2023-11-14T19:49:17.4537520Z Downloading charset_normalizer-3.3.2-cp39-cp39-macosx_10_9_x86_64.whl (122 kB)
2023-11-14T19:49:17.4629340Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 122.5/122.5 kB 17.2 MB/s eta 0:00:00
2023-11-14T19:49:17.4834940Z Downloading PyQt5_Qt5-5.15.11-py3-none-macosx_10_13_x86_64.whl (40.8 MB)
2023-11-14T19:49:19.6910420Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.8/40.8 MB 9.6 MB/s eta 0:00:00
2023-11-14T19:49:19.7034450Z Downloading PyQt5_sip-12.13.0-cp39-cp39-macosx_10_9_universal2.whl (144 kB)
2023-11-14T19:49:19.7713760Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 144.4/144.4 kB 2.1 MB/s eta 0:00:00
2023-11-14T19:49:19.7874640Z Downloading urllib3-2.1.0-py3-none-any.whl (104 kB)
2023-11-14T19:49:19.7989810Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 104.6/104.6 kB 11.0 MB/s eta 0:00:00
2023-11-14T19:49:19.8130340Z Downloading zipp-3.17.0-py3-none-any.whl (7.4 kB)
2023-11-14T19:49:20.0122190Z Building wheels for collected packages: minegauler
2023-11-14T19:49:20.0144110Z   Building wheel for minegauler (pyproject.toml): started
2023-11-14T19:49:20.7569180Z   Building wheel for minegauler (pyproject.toml): finished with status 'done'
2023-11-14T19:49:20.7624210Z   Created wheel for minegauler: filename=minegauler-4.2.0b0-py3-none-any.whl size=1026963 sha256=cbfdebc7214e0c2c672eb11d7ae8f87806f3f1c4e111fe24a44ddc54e2f8d37a
2023-11-14T19:49:20.7630250Z   Stored in directory: /private/var/folders/3s/vfzpb5r51gs6y328rmlgzm7c0000gn/T/pip-ephem-wheel-cache-6xbj3mw1/wheels/04/e1/8c/def6fbfd57b0356272ab5ce982d0b81aaccd5957b4db22561c
2023-11-14T19:49:20.7668700Z Successfully built minegauler
2023-11-14T19:49:20.9871600Z Installing collected packages: zig-minesolver, pytz, PyQt5-Qt5, altgraph, zipp, urllib3, typing-extensions, tabulate, PyQt5-sip, pyinstaller-hooks-contrib, packaging, macholib, idna, charset-normalizer, certifi, attrs, requests, PyQt5, importlib-metadata, requests-toolbelt, pyinstaller, minegauler
2023-11-14T19:49:26.5314090Z Successfully installed PyQt5-5.15.10 PyQt5-Qt5-5.15.11 PyQt5-sip-12.13.0 altgraph-0.17.4 attrs-23.1.0 certifi-2023.7.22 charset-normalizer-3.3.2 idna-3.4 importlib-metadata-6.8.0 macholib-1.16.3 minegauler-4.2.0b0 packaging-23.2 pyinstaller-6.2.0 pyinstaller-hooks-contrib-2023.10 pytz-2023.3.post1 requests-2.31.0 requests-toolbelt-1.0.0 tabulate-0.9.0 typing-extensions-4.8.0 urllib3-2.1.0 zig-minesolver-0.1.1 zipp-3.17.0
2023-11-14T19:49:26.6421560Z 
2023-11-14T19:49:26.6468740Z [notice] A new release of pip is available: 23.0.1 -> 23.3.1
2023-11-14T19:49:26.6570760Z [notice] To update, run: pip install --upgrade pip
2023-11-14T19:49:26.8745800Z ##[group]Run python3 -m scripts.create_release -v -o artefacts/
2023-11-14T19:49:26.8746910Z [36;1mpython3 -m scripts.create_release -v -o artefacts/[0m
2023-11-14T19:49:26.8891960Z shell: /bin/bash -e {0}
2023-11-14T19:49:26.8892580Z env:
2023-11-14T19:49:26.8893210Z   pythonLocation: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-11-14T19:49:26.8894590Z   PKG_CONFIG_PATH: /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/pkgconfig
2023-11-14T19:49:26.8895840Z   Python_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-11-14T19:49:26.8897440Z   Python2_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-11-14T19:49:26.8898540Z   Python3_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-11-14T19:49:26.8899390Z ##[endgroup]
2023-11-14T19:49:27.0292260Z DEBUG:root:Running command: pyinstaller --distpath /var/folders/3s/vfzpb5r51gs6y328rmlgzm7c0000gn/T/pyinstaller-z5kokr9p package/minegauler.spec
2023-11-14T19:49:27.7793500Z 611 INFO: PyInstaller: 6.2.0
2023-11-14T19:49:27.7871430Z 611 INFO: Python: 3.9.18
2023-11-14T19:49:27.8423160Z 671 INFO: Platform: macOS-12.6.9-x86_64-i386-64bit
2023-11-14T19:49:27.9297960Z 754 INFO: Extending PYTHONPATH with paths
2023-11-14T19:49:27.9383590Z ['/Users/runner/work/minegauler/minegauler/package']
2023-11-14T19:49:28.1376040Z 961 INFO: Appending 'datas' from .spec
2023-11-14T19:49:28.1479180Z 967 INFO: checking Analysis
2023-11-14T19:49:28.1513920Z 967 INFO: Building Analysis because Analysis-00.toc is non existent
2023-11-14T19:49:28.1529180Z 967 INFO: Initializing module dependency graph...
2023-11-14T19:49:28.1587330Z 971 INFO: Caching module graph hooks...
2023-11-14T19:49:28.1619530Z 986 INFO: Analyzing base_library.zip ...
2023-11-14T19:49:29.2953660Z 2127 INFO: Loading module hook 'hook-encodings.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:31.5207690Z 4352 INFO: Loading module hook 'hook-pickle.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:32.6480080Z 5480 INFO: Loading module hook 'hook-heapq.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:33.4003860Z 6233 INFO: Caching module dependency graph...
2023-11-14T19:49:33.5667450Z 6388 INFO: Running Analysis Analysis-00.toc
2023-11-14T19:49:33.5688510Z 6388 INFO: Looking for Python shared library...
2023-11-14T19:49:33.5709080Z 6391 INFO: Using Python shared library: /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/libpython3.9.dylib
2023-11-14T19:49:33.5719840Z 6391 INFO: Analyzing /Users/runner/work/minegauler/minegauler/package/.pyinstaller_main.py
2023-11-14T19:49:33.8091940Z 6642 INFO: Loading module hook 'hook-platform.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:33.8499340Z 6682 INFO: Loading module hook 'hook-xml.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:34.0141320Z 6846 INFO: Loading module hook 'hook-importlib_metadata.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:35.8595970Z 8679 INFO: Loading module hook 'hook-charset_normalizer.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/_pyinstaller_hooks_contrib/hooks/stdhooks'...
2023-11-14T19:49:36.5613850Z 9394 INFO: Loading module hook 'hook-certifi.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/_pyinstaller_hooks_contrib/hooks/stdhooks'...
2023-11-14T19:49:36.7960880Z 9624 INFO: Loading module hook 'hook-sqlite3.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:37.6153220Z 10447 INFO: Processing module hooks...
2023-11-14T19:49:37.6189130Z 10451 INFO: Loading module hook 'hook-PyQt5.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:37.7296600Z 10562 WARNING: Hidden import "sip" not found!
2023-11-14T19:49:37.8091490Z 10641 INFO: Loading module hook 'hook-PyQt5.QtGui.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:37.9421100Z 10774 INFO: Loading module hook 'hook-PyQt5.QtWidgets.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:38.1422140Z 10972 INFO: Loading module hook 'hook-PyQt5.QtCore.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:38.4270640Z 11259 INFO: Looking for ctypes DLLs
2023-11-14T19:49:38.4364140Z 11266 INFO: Analyzing run-time hooks ...
2023-11-14T19:49:38.4402100Z 11271 INFO: Including run-time hook '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks/rthooks/pyi_rth_inspect.py'
2023-11-14T19:49:38.4407840Z 11273 INFO: Including run-time hook '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks/rthooks/pyi_rth_pyqt5.py'
2023-11-14T19:49:38.4440810Z 11276 INFO: Processing pre-find module path hook _pyi_rth_utils from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks/pre_find_module_path/hook-_pyi_rth_utils.py'.
2023-11-14T19:49:38.4460330Z 11277 INFO: Loading module hook 'hook-_pyi_rth_utils.py' from '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks'...
2023-11-14T19:49:38.4485700Z 11278 INFO: Including run-time hook '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/hooks/rthooks/pyi_rth_pkgutil.py'
2023-11-14T19:49:38.4788880Z 11310 INFO: Looking for dynamic libraries
2023-11-14T19:49:38.9890860Z 11821 WARNING: Library not found: could not resolve '@rpath/QtVirtualKeyboard.framework/Versions/5/QtVirtualKeyboard', dependency of '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyQt5/Qt5/plugins/platforminputcontexts/libqtvirtualkeyboardplugin.dylib'.
2023-11-14T19:49:38.9900370Z 11821 WARNING: Library not found: could not resolve '@rpath/QtPdf.framework/Versions/5/QtPdf', dependency of '/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyQt5/Qt5/plugins/imageformats/libqpdf.dylib'.
2023-11-14T19:49:39.0789850Z 11912 INFO: Warnings written to /Users/runner/work/minegauler/minegauler/build/minegauler/warn-minegauler.txt
2023-11-14T19:49:39.1283530Z 11959 INFO: Graph cross-reference written to /Users/runner/work/minegauler/minegauler/build/minegauler/xref-minegauler.html
2023-11-14T19:49:39.1318310Z Traceback (most recent call last):
2023-11-14T19:49:39.1366030Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/bin/pyinstaller", line 8, in <module>
2023-11-14T19:49:39.1367540Z     sys.exit(_console_script_run())
2023-11-14T19:49:39.1369730Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/__main__.py", line 214, in _console_script_run
2023-11-14T19:49:39.1371930Z     run()
2023-11-14T19:49:39.1392560Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/__main__.py", line 198, in run
2023-11-14T19:49:39.1462890Z     run_build(pyi_config, spec_file, **vars(args))
2023-11-14T19:49:39.1495430Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/__main__.py", line 69, in run_build
2023-11-14T19:49:39.1518330Z     PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
2023-11-14T19:49:39.1521200Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 1069, in main
2023-11-14T19:49:39.1523490Z     build(specfile, distpath, workpath, clean_build)
2023-11-14T19:49:39.1527350Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 1009, in build
2023-11-14T19:49:39.1529870Z     exec(code, spec_namespace)
2023-11-14T19:49:39.1530890Z   File "package/minegauler.spec", line 31, in <module>
2023-11-14T19:49:39.1533930Z     app_analysis = Analysis(
2023-11-14T19:49:39.1566850Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 470, in __init__
2023-11-14T19:49:39.1578460Z     self.__postinit__()
2023-11-14T19:49:39.1594630Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/building/datastruct.py", line 184, in __postinit__
2023-11-14T19:49:39.1606530Z     self.assemble()
2023-11-14T19:49:39.1611310Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 852, in assemble
2023-11-14T19:49:39.1618960Z     sdk_version = osxutils.get_macos_sdk_version(src_name)
2023-11-14T19:49:39.1632220Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/utils/osx.py", line 113, in get_macos_sdk_version
2023-11-14T19:49:39.1637170Z     version_cmd = _find_version_cmd(header)
2023-11-14T19:49:39.1647120Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/PyInstaller/utils/osx.py", line 98, in _find_version_cmd
2023-11-14T19:49:39.1652130Z     assert len(version_cmd) == 1, "Expected exactly one LC_BUILD_VERSION or LC_VERSION_MIN_MACOSX command!"
2023-11-14T19:49:39.1654880Z AssertionError: Expected exactly one LC_BUILD_VERSION or LC_VERSION_MIN_MACOSX command!
2023-11-14T19:49:39.2131850Z Traceback (most recent call last):
2023-11-14T19:49:39.2162150Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/runpy.py", line 197, in _run_module_as_main
2023-11-14T19:49:39.2180010Z     return _run_code(code, main_globals, None,
2023-11-14T19:49:39.2246510Z   File "/Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/runpy.py", line 87, in _run_code
2023-11-14T19:49:39.2273280Z     exec(code, run_globals)
2023-11-14T19:49:39.2320460Z   File "/Users/runner/work/minegauler/minegauler/scripts/create_release.py", line 218, in <module>
2023-11-14T19:49:39.2390030Z     main(sys.argv[1:])
2023-11-14T19:49:39.2396690Z   File "/Users/runner/work/minegauler/minegauler/scripts/create_release.py", line 212, in main
2023-11-14T19:49:39.2440420Z     run_pyinstaller(tmpdir)
2023-11-14T19:49:39.2444620Z   File "/Users/runner/work/minegauler/minegauler/scripts/create_release.py", line 72, in run_pyinstaller
2023-11-14T19:49:39.2461410Z     raise Exception(f"Pyinstaller command with exit code {proc.returncode}")
2023-11-14T19:49:39.2462580Z Exception: Pyinstaller command with exit code 1
```

Note that x86_64 may not be the right architecture choice for MacOS now (in README)?

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -12,8 +12,8 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ubuntu-latest, windows-latest]
-#        os: [ubuntu-latest, macOS-latest, windows-latest]
+        # macOS-latest is arm64, macOS-13 is x86_64
+        os: [ubuntu-latest, windows-latest, macOS-latest, macOS-13]
 
     steps:
       - uses: actions/checkout@master
@@ -21,16 +21,15 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@master
         with:
-          python-version: "3.9"
+          python-version: "3.12"
 
       - name: Install Python requirements
         run: |
-          curl -LsSf https://astral.sh/uv/install.sh | sh
-          uv sync
-          echo "$GITHUB_WORKSPACE/.venv/bin" >> "$GITHUB_PATH"
+          pip install uv
+          uv sync --no-editable
 
       - name: Create PyInstaller package
-        run: python3 -m scripts.create_release -v -o artefacts/
+        run: uv run -m scripts.create_release -v -o artefacts/
 
       - name: Store artefacts
         uses: actions/upload-artifact@master

--- a/README.md
+++ b/README.md
@@ -21,10 +21,10 @@ Read more about the project history in [my blog post](https://www.lewisgaul.co.u
 The application has been packaged with PyInstaller so that it can be played without setting up Python.
 
 Download links available here:
- - [Windows (amd64)](https://github.com/LewisGaul/minegauler/releases/download/v4.2.0-b0/minegauler-v4.2.0-b0-windows-amd64.zip)
- - [Linux (x86_64)](https://github.com/LewisGaul/minegauler/releases/download/v4.2.0-b0/minegauler-v4.2.0-b0-linux-x86_64.tar.gz)
- - [MacOS (x86_64)](https://github.com/LewisGaul/minegauler/releases/download/v4.2.0-b0/minegauler-v4.2.0-b0-macos-x86_64.tar.gz)
- - MacOS (arm64) (not yet available, see [issue #214](https://github.com/LewisGaul/minegauler/issues/214))
+ - [Windows (amd64)](https://github.com/LewisGaul/minegauler/releases/download/v4.2.0/minegauler-v4.2.0-windows-amd64.zip)
+ - [Linux (x86_64)](https://github.com/LewisGaul/minegauler/releases/download/v4.2.0/minegauler-v4.2.0-linux-x86_64.tar.gz)
+ - [MacOS (arm64)](https://github.com/LewisGaul/minegauler/releases/download/v4.2.0/minegauler-v4.2.0-macos-arm64.tar.gz)
+ - [MacOS (x86_64)](https://github.com/LewisGaul/minegauler/releases/download/v4.2.0/minegauler-v4.2.0-macos-x86_64.tar.gz)
 
 
 ### Install from PyPI

